### ISSUE TITLE
Are you still active in this community?

### ISSUE THREAD

**Issue by strich** [tostercx/eve_probe#9]
I imagine you've since moved on from this particular project, but I'd love to pick your brain about it if you're open to it?

**Comment by tostercx** [comment]
Haven't touched this (or eve) in a long time. I doubt I'd be of much help but sure.

**Comment by strich** [comment]
@tostercx did you ever get the probe scanner window automated? In particular the probe cursor so it can be placed on to sigs for scanning.

**Comment by tostercx** [comment]
Yeah, that was probably the only script that worked somewhat well. It just scanned all anomalies automatically once the probes were released IIRC.

The project was primarily a GUI network packet inspector/editor/injector. The scripts are mostly an afterthought testing out what can be done with it. There's probably an easier way to automate scanning with the client than composing packets if that's what you're after ^^;

**Comment by strich** [comment]
Yeah ideally I don't use the packet approach hah. Could you tell me a little more about the process to hook the python domain and, I assume [what you're doing here](https://github.com/tostercx/eve_probe/blob/611c9f5f40822a2dcc3acc85da19ba5560c87663/eve_probe/py/scan.py#L30), is inject some custom functions that you can call from outside EVE.

**Comment by tostercx** [comment]
So... I am using eve's python, but it's not the same instance as the game is running in. The main application hooks, captures and decrypts raw packets from a running eve process. It then loads eve's python in my app to parse the packet and convert it to a dumb object that can be represented as JSON for human viewing/editing.

At the spot you mentioned I'm creating a new packet in the human viewable format.

I know the architecture is confusing but the only interaction I have with the game client is reading or writing packets.

**Comment by strich** [comment]
Ahhh. I see. That's very interesting! I'm half tempted to maybe get it working again just coz that's a kind of cool method you've developed. But as you say I wouldn't plan to expanding it if I can avoid it.
I suppose you are at least calling a real EVE python method to at least send your packets right? Or are constructing and sending the packet yourself entirely outside of the EVE app?

**Comment by tostercx** [comment]
Well my app uses eve's libraries to make the packet and convert it to a binary buffer from the outside (blue.marshal.Load/Save). It then passes it to eve's process to send it. I'm only hooking CryptEncrypt and WSA Send and friends from the Win32 API.

As for getting it working - I'm not sure if the traffic capture part still works. May need some debugging.

If you want to run python directly in the game client, I think you can get the address of `PyRun_SimpleString` and pass it a string. It had to be run from the main thread or it crashed due to stackless python weirdness. Not sure if that still works, they did block some of the easier ways to run code in the past.
